### PR TITLE
Dev/kpf arrow

### PR DESCRIPTION
**Pull request by collinsr**
This branch is mostly Aaron's work moving KPF out of vital and into an arrow.

### CODE REVIEW

**Comment by aaron-bray** [comment]
LGTM

File: arrows/kpf/yaml/CMakeLists.txt [hunk 1]
```diff
@@ -42,19 +38,19 @@ set( private_sources
...
 
-kwiver_add_library( vital_kpf
+kwiver_add_library( kpf_yaml
   ${sources}
   ${public_headers}
   ${private_sources}
 )
 
-target_include_directories(vital_kpf PRIVATE ${YAML_CPP_INCLUDE_DIR})
+target_include_directories(kpf_yaml PRIVATE ${YAML_CPP_INCLUDE_DIR})
```
Comment (dstoup) [inline arrows/kpf/yaml/CMakeLists.txt]: If kfp_yaml is including <yaml-cpp/yaml.h> in any of its headers, this include_directories needs to be PUBLIC.

File: examples/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -4,7 +4,28 @@ set(SOURCE "main.cpp"
...
+                     kpf/yaml_parser.cxx
+                     kpf/kpf_example_yaml.cxx
+  )
+
+  list(APPEND kwiver_examples_libraries kwiver_algo_kpf kpf_yaml ${YAML_CPP_LIBRARIES})
+  if(WIN32)
+    set(kwiver_examples_flags YAML_CPP_DLL)
+  endif()
+endif()
+
```
Comment (dstoup) [inline examples/cpp/CMakeLists.txt]: The target(s) in here which include <yaml-cpp/yaml.h> need access to the include_dirs for yaml. You will need to copy the target_include_dirs call from the arrows/kpf CMakeLists.txt here, or change that one so it's just include_directories and move it somewhere that allows scoping to carry the include_directories everywhere it's needed.

**Comment by linus-sherrill** [comment]
Jenkins test this please

**Comment by collinsr** [comment]
Jenkins test this please

File: arrows/kpf/yaml/CMakeLists.txt (hunk 1 above)
Comment (collinsr) [reply arrows/kpf/yaml/CMakeLists.txt]: Done!

**Comment by aaron-bray** [comment]
The new commits looks good
Sorry the single example executable didn't work and you had to undo it
It probably is better as independent examples, as they are more 'low level' that vital proper interfaces

**Comment by linus-sherrill** [comment]
There is a problem in windows - kpf_exception (and related classes) may have the wrong EXPORT symbol

**Comment by collinsr** [comment]
d'oh-- fixed

**Comment by linus-sherrill** [comment]
Jenkins test this please

**Comment by linus-sherrill** [comment]
Jenkins test this please

**Comment by dstoup** [comment]
@collinsr It looks like that tentative fix for travis was the right thing to do. We are now failing for a different reason.

**Comment by collinsr** [comment]
looks like clang builds / installs / 4 tests fail, gcc builds / fails to install due to missing sprokit config?

**Comment by dstoup** [comment]
Branch looks good to me. The travis test failures exist on master. This branch actually only has 13 to the Jenkins' master's 15

**Pull request merged**